addons/account/models/account_move.py

```diff
@@ -1556,6 +1564,37 @@ def _fetch_duplicate_supplier_reference(self, only_posted=False):
             casted_values = ", ".join([f"%({field_name})s::{moves._fields[field_name].column_type[0]}" for field_name in place_holders])
             move_table_and_alias = f'(VALUES ({casted_values})) AS move({", ".join(place_holders)})'
 
+        if move_type == 'out_invoice':
+            self._query_duplicate_out_invoice(move_table_and_alias, moves, place_holders)
+        else:
+            self._query_duplicate_in_invoice(move_table_and_alias, moves, only_posted, place_holders)
+        return {
+            self.env['account.move'].browse(res['move_id']): self.env['account.move'].browse(res['duplicate_ids'])
+            for res in self.env.cr.dictfetchall()
+        }
+
+    def _query_duplicate_out_invoice(self, move_table_and_alias, moves, place_holders):
+        self.env.cr.execute(f"""
+            SELECT
+                   move.id AS move_id,
+                   array_agg(duplicate_move.id) AS duplicate_ids
+              FROM {move_table_and_alias}
+              JOIN account_move AS duplicate_move ON
+                   move.company_id = duplicate_move.company_id
+               AND move.amount_total = duplicate_move.amount_total
+               AND move.move_type = duplicate_move.move_type
+               AND move.id != duplicate_move.id
+               AND move.invoice_date = duplicate_move.invoice_date
+               AND duplicate_move.state != 'cancel'
+               AND duplicate_move.state = 'posted'
```

Comment:
Just putting the conditions into an if-else and injecting it should make the job.
The function already contains a bigger sin anyway.